compiler/rustc_error_codes/src/error_codes/E0311.md

```diff
@@ -0,0 +1,52 @@
+This error occurs when there is an unsatisfied outlives bound involving an
+elided region on a generic type parameter or associated type.
+
+Erroneous code example:
+
+```compile_fail,E0311
+fn no_restriction<T>(x: &()) -> &() {
+    with_restriction::<T>(x)
+}
+
+fn with_restriction<'a, T: 'a>(x: &'a ()) -> &'a () {
+    x
+}
+```
+
+Why doesn't this code compile? It helps to look at the lifetime bounds that are
+automatically adding by the compiler. For more details see the Rust
+Documentation for Lifetime Elision:
+https://doc.rust-lang.org/reference/lifetime-elision.html.
```

Comment:
Are there any best practices around how to include links like this in the documentation? For example, would this be better?

```markdown
For more details see the
[Rust Documentation for Lifetime Elision]
(https://doc.rust-lang.org/reference/lifetime-elision.html)
```